src/scoped.rs

```diff
@@ -78,6 +94,116 @@ impl<'a> Scope<'a> {
         });
     }
 
+    /// Create a scoped thread.
+    ///
+    /// `spawn` is similar to the [`spawn`][spawn] function in Rust's standard library. The
+    /// difference is that this thread is scoped, meaning that it's guaranteed to terminate
+    /// before the current stack frame goes away, allowing you to reference the parent stack frame
```

Comment:
Can you mention that this guarantee is ensured by having the parent thread join on the child thread before exiting the scope?